consensus/consensus.go

```diff
@@ -1,65 +1,180 @@
 package consensus
 
 import (
+	"errors"
 	"time"
 
 	"github.com/deso-protocol/core/bls"
 )
 
 func NewFastHotStuffConsensus() *FastHotStuffConsensus {
 	return &FastHotStuffConsensus{
-		internalTimersUpdated: make(chan interface{}),
-		votesSeen:             make(map[BlockHash]map[bls.PublicKey]*bls.Signature),
-		timeoutsSeen:          make(map[BlockHash]map[bls.PublicKey]*bls.Signature),
-		ConsensusEvents:       make(chan *ConsensusEvent),
+		status: consensusStatusNotInitialized,
 	}
 }
 
-func (fc *FastHotStuffConsensus) Init( /*TODO */ ) {
+// Initializes the consensus instance with the latest known valid block in the blockchain, and
+// the validator set for the next block height. The functions expects the following for the input
+// params:
+//   - Block construction duration must be > 0
+//   - Timeout base duration must be > 0
+//   - The input block must have a valid block hash, block height, view, and QC
+//   - The validators param must be sorted in decreasing order of stake, with a
+//     consistent tie breaking scheme. The validator set is expected to be valid for
+//     validating votes and timeouts for the next block height.
+//
```

Comment:
> //   - Block construction duration must be > 0
 //   - Timeout base duration must be > 0
 //   - The input block must have a valid block hash, block height, view, and QC
 //   - The validators param must be sorted in decreasing order of stake, with a
 //     consistent tie breaking scheme. The validator set is expected to be valid for
 //     validating votes and timeouts for the next block height.
 //


Maybe we instead make it:


```
 //   - blockConstructionCadence: block construction duration must be > 0
 //   - timeoutBaseDuration: base duration must be > 0
 //   - chainTip: The input block must have a valid block hash, block height, view, and QC
 //   - validators: The validators param must be sorted in decreasing order of stake, with a
 //     consistent tie breaking scheme. The validator set is expected to be valid for
 //     validating votes and timeouts for the next block height.
 //

```

Or something like this, so it's more clear.